drivers/ieee802154/ieee802154_cc1101.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2021 Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc1101
+
+#define LOG_MODULE_NAME ieee802154_cc1101
+#define LOG_LEVEL CONFIG_IEEE802154_DRIVER_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <errno.h>
+
+#include <kernel.h>
+#include <arch/cpu.h>
+
+#include <device.h>
+#include <init.h>
+#include <net/net_if.h>
+#include <net/net_pkt.h>
+
+#include <sys/byteorder.h>
+#include <string.h>
+#include <random/rand32.h>
+
+#include <drivers/rf.h.h>
+#include <net/ieee802154_radio.h>
+
+#include "ieee802154_cc1101.h"
+
+
+
+static inline uint8_t *get_mac(const struct device *dev)
+{
+	struct cc1101_context *cc1101 = dev->data;
+
+#if defined(CONFIG_IEEE802154_CC1200_RANDOM_MAC)
+	uint32_t *ptr = (uint32_t *)(cc1101->mac_addr + 4);
+
+	UNALIGNED_PUT(sys_rand32_get(), ptr);
+
+	cc1100->mac_addr[7] = (cc1200->mac_addr[7] & ~0x01) | 0x02;
+#else
+	cc1101->mac_addr[4] = CONFIG_IEEE802154_CC1101_MAC4;
+	cc1101->mac_addr[5] = CONFIG_IEEE802154_CC1101_MAC5;
+	cc1101->mac_addr[6] = CONFIG_IEEE802154_CC1101_MAC6;
+	cc1101->mac_addr[7] = CONFIG_IEEE802154_CC1101_MAC7;
+#endif
+
+	cc1101->mac_addr[0] = 0x00;
+	cc1101->mac_addr[1] = 0x12;
+	cc1101->mac_addr[2] = 0x4b;
+	cc1101->mac_addr[3] = 0x00;
+
+	return cc1101->mac_addr;
+}
+
+static int cc1101_get_capabilities(const struct device *dev)
+{
+	return IEEE802154_HW_SUB_GHZ;
+}
+
+static int cc1101_cca(const struct device *dev)
+{
+	int ret;
```

Comment:
ret seems superfluous here, just return directly from the RF API call

same in various places below